### PR TITLE
Doc metric collection

### PR DESCRIPTION
**Pull request by vgratian**
Address issue #37 

* Add uniform structure to docs of collectors
* Explain metric collection/calculation paradigm of each collector
* Fix a few typos

### CODE REVIEW

**Comment by vgratian** [comment]
+ removed `allow_addrs_regex` from influxDB's README, was there by mistake

File: cmd/collectors/unix/README.md [hunk 1]
```diff
@@ -1,13 +1,21 @@
 
 
-# Unix Collector
+# Unix
 
-## Overview
+This collector polls resource usage by Harvest pollers on the local system. Collector might be extended in the future to monitor any local or remote process.
 
-This collector polls resource usage by Harvest pollers on the local system.
+### Table of Contents
```
Comment (cgrinds) [inline cmd/collectors/unix/README.md:7]: Do you think TOC is worth adding and more importantly, maintaining when GitHub automatically creates one for md files already?

File: cmd/collectors/zapi/README.md [hunk 2]
```diff
@@ -1,11 +1,24 @@
 
 
-# Zapi Collector
+# Zapi
 
-Zapi collects data from ONTAP systems using the ZAPI protocol. The collector submits data as it is and does not perform any calculations (therefore it is not able to collect `perf` objects). Since the attributes of most APIs have an irregular tree structure, sometimes a plugin will be required to collect metrics from an API.
+Zapi collects data from ONTAP systems using the ZAPI protocol. The collector submits data as received from the target system, and does not perform any calculations or postprocessng. Since the attributes of most APIs have an irregular tree structure, sometimes a plugin will be required to collect all metrics from an API.
```
Comment (cgrinds) [inline cmd/collectors/zapi/README.md:5]: postprocessng should be post-processing

File: cmd/collectors/zapi/README.md [hunk 3]
```diff
@@ -1,11 +1,24 @@
...
 
+Note that the [ZapiPerf collector](../zapiperf/README.md) is an extension of this collector, therefore many parameters and configuration settings will coincide.
 
-## Configuration
+### Table of Contents
```
Comment (cgrinds) [inline cmd/collectors/zapi/README.md:9]: Same TOC comment

File: cmd/collectors/zapiperf/README.md [hunk 4]
```diff
@@ -1,10 +1,24 @@
-# ZapiPerf Collector
+# ZapiPerf
 
 ZapiPerf collects performance metrics from ONTAP systems using the ZAPI protocol. The collector is designed to be easily extendible to collect new objects or to collect additional counters from already configured objects. (The [default configuration](../../../conf/zapiperf/default.yaml) file contains 25 objects)
 
-Note: The main difference between this collector and the Zapi collector is that ZapiPerf collects only the `perf` subfamily of the ZAPIs. Additionally, ZapiPerf always calculates final values from deltas of two subsequent polls (therefore data is emitted only after the second poll).
+This collector is an extension of the [Zapi collector](../zapi/README.md) with the major difference between that ZapiPerf collects only the `perf` subfamily of the ZAPIs. Additionally, ZapiPerf always calculates final values from deltas of two subsequent polls.
+
+
+### Table of Contents
```
Comment (cgrinds) [inline cmd/collectors/zapiperf/README.md:8]: Same TOC comment

**Review by cgrinds — COMMENTED** [review]
Thanks for the additional details @vgratian. 
Few minor comments below.

File: cmd/collectors/unix/README.md (hunk 1 above)
Comment (vgratian) [reply cmd/collectors/unix/README.md:7]: haven't noticed auto-generated TOC, but the TOC is not critical, what I suggest is to have a (non-strict,) uniform structure for all collectors, to make it easier to find the info you need. e.g. #Metric should provide what metrics the collector collects and how (if any) final values are calculated)

File: cmd/collectors/zapi/README.md [hunk 5]
```diff
@@ -2,16 +2,10 @@
 
 # Zapi
 
-Zapi collects data from ONTAP systems using the ZAPI protocol. The collector submits data as received from the target system, and does not perform any calculations or postprocessng. Since the attributes of most APIs have an irregular tree structure, sometimes a plugin will be required to collect all metrics from an API.
+Zapi collects data from ONTAP systems using the ZAPI protocol. The collector submits data as received from the target system, and does not perform any calculations or post-processng. Since the attributes of most APIs have an irregular tree structure, sometimes a plugin will be required to collect all metrics from an API.
```
Comment (cgrinds) [inline cmd/collectors/zapi/README.md:5]: post-processing instead of post-processng.

File: cmd/collectors/unix/README.md (hunk 1 above)
Comment (cgrinds) [reply cmd/collectors/unix/README.md]: The auto-TOC is near the top of the file
image

**Review by cgrinds — APPROVED** [review]
Nice update. Thanks @vgratian

**Pull request merged**